db/seeds.rb

```diff
@@ -47,12 +47,12 @@
 
 if Region.all.empty?
   puts 'Loading regions'
-  Rake::Task['shf:load_regions'].invoke
+  Rake::Task['shf:load_regions'].execute
```

Comment:
You're explicitly calling the task `shf:load_regions` with the `invoke` method, so it will be run.
I don't understand your comment "#invoke only executes when needed, which would be only once for all tests)"


The major difference between `invoke` and `execute` is about _dependencies:_

  - `invoke` will ensure that all task dependencies will be run 
  - `execute` does _not_ run task dependencies

So I think that in this case it is better to use `invoke` in case the rake task `shf:load_regions` has other dependencies.

Here's a little bit of rake code that will show the difference:

```
 

  desc 'task1 no dependencies'
  task :task1 do | _t |
    puts "in #{_t.name}"
  end

  desc 'task2 depends on task1'
  task :task2 => :task1 do | _t |
    puts "in #{_t.name}"
  end

  desc 'task3 invoke v. execute task2'
  task :task3 do | _t |
    task2_name = 'shf:task2'

    puts "INVOKE task2:"
    Rake::Task[task2_name].invoke

    puts "EXECUTE task2:"
    Rake::Task[task2_name].execute
  end

```

Results:
```
INVOKE task2:
in shf:task1
in shf:task2
EXECUTE task2:
in shf:task2
```